Tensile/KernelWriterAssembly.py

```diff
@@ -12047,31 +12119,34 @@ def globalWriteBatch(self, kernel, ss, batchIdx, applyAlpha, beta, edge, atomic,
         # first attempt write
         kStr += self.comment("issue first atomic writes")
         for elementIdx in range(0, len(batchElements)):
-          element  = batchElements[elementIdx]
           addrCalc = ss.elementAddr[elementIdx]
           mask     = ss.elementMask[elementIdx]
-          d1       = element[0]
-          d0       = element[1]
-          vc1      = element[2]
-          vc0      = element[3]
-
-          # apply in-bounds exec mask
-          if edge:
-            kStr += inst("s_mov_b{}".format(wavelen), self.exec, sgpr(mask,laneSGPRC), "sgprs -> exec (before atomic)" )
 
-          for avi in range(0, gwvw//atomicW):
-            dataV = ss.elementData[elementIdx] + int(avi*ss.cfg.numVgprsPerDataPerVI)
-            sumIdxV = ss.elementSumIdx[elementIdx] + avi
-            if self.do["GlobalWrite"]:
-              if kernel["BufferStore"]:
-                kStr += "buffer_atomic_add_f32 %s, %s, %s, %s    // %s%s" % \
-                    (vgpr("ValuC+%u"%sumIdxV), \
-                     vgpr(addrCalc.addrDVgpr,1), \
-                     sgpr("SrdD", 4), \
-                     "0 offen offset:%u" % addrCalc.globalOffset, \
-                     "attempt write avi=%u" % (avi), self.endLine )
-              else:
-                pass # TODO:
+          loopCount = 0
+          for avi in range(0, gwvw, atomicW):
+            for compIdx in range(vgprLoadDW // atomicOpW):
+              maskIdx = mask + loopCount * laneSGPRC
```

Comment:
What does loopCount do here? Looks like it never gets updated

Reply:
> What does loopCount do here? Looks like it never gets updated

Thanks for pointing this out.
I changed atomicAdd part later and forgot to add increment code.
I will add it.

Reply:
> > What does loopCount do here? Looks like it never gets updated
> 
> Thanks for pointing this out. I changed atomicAdd part later and forgot to add increment code. I will add it.

It looks like just incrementing the counter is not enough.
After I debugged it, it turned out that avi was always 0 regardless of StoreVectorWidth setting due to the restriction in maxGwvw.
When I removed the restriction, the test failed with avi>1 cases.
I will fix the issue and provide update later.

Reply:
> > > What does loopCount do here? Looks like it never gets updated
> > 
> > 
> > Thanks for pointing this out. I changed atomicAdd part later and forgot to add increment code. I will add it.
> 
> It looks like just incrementing the counter is not enough. After I debugged it, it turned out that avi was always 0 regardless of StoreVectorWidth setting due to the restriction in maxGwvw. When I removed the restriction, the test failed with avi>0 cases. I will fix the issue and provide update later.

I fixed all issues and enabled avi>0 cases.
The current version should be OK.